lib/floe/workflow/states/fail.rb

```diff
@@ -13,8 +13,13 @@ def initialize(workflow, name, payload)
           @error = payload["Error"]
         end
 
-        def run!(input)
-          [nil, input]
+        def run_async!(input)
+          context.next_state = nil
+          context.output     = input
```

Comment:
need to set cause and error

Reply:
I had that in finish, though not sure why and it works just as well here and I can drop the `finish` override